openhands/llm/fn_call_converter.py

```diff
@@ -798,22 +800,23 @@ def convert_non_fncall_messages_to_fncall_messages(
                     isinstance(content, str)
                     or (
                         isinstance(content, list)
-                        and len(content) == 1
-                        and content[0].get('type') == 'text'
+                        and first_tool_call_content
```

Comment:
first_tool_call_content is not defined before, i imagine we will run into issue for this?

i.e., shall we define `first_tool_call_content = None` before enter into this if-else clauses